pkg/server/server.go

```diff
@@ -446,7 +446,22 @@ func cmdDelegateAdd(cmdArgs *skel.CmdArgs, k8sArgs *types.K8sArgs, exec invoke.E
 		return nil, err
 	}
 
-	delegateCNIConf, err := types.LoadDelegateNetConf(cmdArgs.StdinData, nil, "", "")
+	// copy deleate annotation into network selection element
+	var selectionElement *types.NetworkSelectionElement
+	if annotation != nil {
+		selectionElement = &types.NetworkSelectionElement{}
+		if annotation.MacRequest != "" {
+			selectionElement.MacRequest = annotation.MacRequest
+		}
+		if annotation.IPRequest != nil {
+			selectionElement.IPRequest = annotation.IPRequest
+		}
+		if annotation.CNIArgs != nil {
+			selectionElement.CNIArgs = annotation.CNIArgs
+		}
+	}
+
+	delegateCNIConf, err := types.LoadDelegateNetConf(cmdArgs.StdinData, selectionElement, "", "")
```

Comment:
nit: I would define an help function here, something like:
```golang
delegateCNIConf, err := types.LoadDelegateNetConf(cmdArgs.StdinData, delegateNetworkSelectionElement(annotation), "", "")
...

func delegateNetworkSelectionElement(interfaceAttributes *api.DelegateAnnotatio) *types.NetworkSelectionElement {
    if annotation != nil {
		selectionElement = &types.NetworkSelectionElement{}
		if annotation.MacRequest != "" {
			selectionElement.MacRequest = annotation.MacRequest
		}
		if annotation.IPRequest != nil {
			selectionElement.IPRequest = annotation.IPRequest
		}
		if annotation.CNIArgs != nil {
			selectionElement.CNIArgs = annotation.CNIArgs
		}
                return selectionElement
	}
    return nil
}
```

Reply:
Currently I don't come up with to reuse this code yet, so let me keep it. Once we reuse it then let's make helper function.